Guides/Modeling and Binding.md

```diff
@@ -0,0 +1,584 @@
+  case is ImageViewModel: height = 250
+  case is Comment: height = 35
+  // 3
+  default: height = 55
+  }
+  return CGSize(width: width, height: height)
+}
+```
+
+1. Just like the `object` property, the `collectionContext` should never be `nil`, but it's a weakly referenced object so must be declared as optional. Again use `fatalError()` to catch any critical failures.
+2. Swift makes checking for types so easy! Just `switch` on the type and assign a height. In Objective-C you should use `isKindOfClass:`.
+3. Both the `UserViewModel` and `ActionViewModel` share the same height of `55`pts according to the design.
+
+Lastly, implement the API that returns a cell for each view model. This should look similar to the size API above. Give it a try yourself.
+
+> Remember that the cells are defined in **Main.storyboard**. You can click on each cell to view their identifiers.
+
+<details>
+  <summary>"cellForViewModel:" implementation</summary>
+  <p>
+
+```swift
+func sectionController(
+  _ sectionController: ListBindingSectionController<ListDiffable>,
+  cellForViewModel viewModel: Any,
+  at index: Int
+  ) -> UICollectionViewCell {
+  let identifier: String
+  switch viewModel {
+  case is ImageViewModel: identifier = "image"
+  case is Comment: identifier = "comment"
+  case is UserViewModel: identifier = "user"
+  default: identifier = "action"
+  }
+  guard let cell = collectionContext?
+    .dequeueReusableCellFromStoryboard(withIdentifier: identifier, for: self, at: index)
+    else { fatalError() }
+  return cell
+}
+```
+
+Remember to handle `UserViewModel` and `ActionViewModel` separately!
+
+</p></details>
+
+## Binding Models to Cells
+
+Now you have `PostSectionController` setup to create view models, sizes, and cells. The last piece to using `IGListBindingSectionController` is having your cells to receive its assigned view model and configure itself. 
+
+This is done by making your cells conform to `IGListBindable`. With that, `IGListBindingSectionController` will **automatically** bind view models to each cell!
+
+Open **ImageCell.swift** and change the implementation to look like the following:
+
+```swift
+import UIKit
+import SDWebImage
+// 1
+import IGListKit
+
+// 2
+final class ImageCell: UICollectionViewCell, ListBindable {
+
+  @IBOutlet weak var imageView: UIImageView!
+
+  // MARK: ListBindable
+
+  func bindViewModel(_ viewModel: Any) {
+    // 3
+    guard let viewModel = viewModel as? ImageViewModel else { return }
+    // 4
+    imageView.sd_setImage(with: viewModel.url)
+  }
+
+}
+```
+
+1. Make sure to import `IGListKit`!
+2. Have the cell conform to `ListBindable`
+3. Guard against the view model type. This will always be what `PostSectionController` decides, but guarding is a safer practive than force casting.
+4. Use the [SDWebImage](https://github.com/rs/SDWebImage) library to set the image URL.
+
+Now do exactly the same thing for each of the other cells:
+
+- `ActionCell` binds `ActionViewModel`
```

Comment:
You are repeating yourself here ;) I think you meant t fallout the other models?